SecureSubmit.php

```diff
@@ -2013,44 +2023,26 @@ function submit_payment() {
                 }
             }
 
-            $config = new HpsServicesConfig();
-
-            $config->secretApiKey = esc_attr($skey);
-            $config->versionNumber = '1648';
-            $config->developerId = '002914';
-
-            $chargeService = new HpsCreditService($config);
-
-            $address = new HpsAddress();
+            $config = $this->setConfig($skey);
+            $address = new Address();
             $address->address = $billing_address;
             $address->city = $billing_city;
             if ($requireState)
                 $address->state = $billing_state;
             $address->zip = $billing_zip;
-
-            $cardHolder = new HpsCardHolder();
-            $cardHolder->firstName = $billing_firstname;
-            $cardHolder->lastName = $billing_lastname;
-            $cardHolder->email = $billing_email;
-            $cardHolder->address = $address;
-
-            $cardOrToken = new HpsTokenData();
-            $cardOrToken->tokenValue = $secureToken;
-
+   
+            $cardOrToken = new CreditCardData();
+            $cardOrToken->token = $secureToken; 
+            $cardOrToken->cardHolderName = $billing_firstname;
```

Comment:
First name and last name should be included here.

Reply:
updated